src/backends/sentry_backend_inproc.c

```diff
@@ -455,6 +455,53 @@ registers_from_uctx(const sentry_ucontext_t *uctx)
     return registers;
 }
 
+#ifdef SENTRY_PLATFORM_LINUX
+static uintptr_t
+get_stack_pointer(const sentry_ucontext_t *uctx)
+{
+#    if defined(__i386__)
+    return uctx->user_context->uc_mcontext.gregs[REG_ESP];
+#    elif defined(__x86_64__)
+    return uctx->user_context->uc_mcontext.gregs[REG_RSP];
+#    elif defined(__arm__)
+    return uctx->user_context->uc_mcontext.arm_sp;
+#    elif defined(__aarch64__)
+    return uctx->user_context->uc_mcontext.sp;
+#    elif defined(__mips__)
+    return uctx->user_context->uc_mcontext.gregs[29]; // REG_SP
+#    elif defined(__riscv)
+    return uctx->user_context->uc_mcontext.__gregs[2]; // REG_SP
+#    else
+    SENTRY_WARN("get_stack_pointer is not implemented for this architecture. "
+                "Signal chaining may not work as expected.");
+    return NULL;
```

Comment:
is `#error` or a runtime warning a better choice here? i ended up with a warning because i thought it would be annoying to potentially break builds for irrelevant platforms that don't even support .NET